whois-api/src/main/java/net/ripe/db/whois/api/TopLevelFilter.java

```diff
@@ -0,0 +1,50 @@
+package net.ripe.db.whois.api;
+
+import com.google.common.collect.Maps;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.etree.NestedIntervalMap;
+import net.ripe.db.whois.common.ip.Interval;
+import net.ripe.db.whois.common.ip.IpInterval;
+import net.ripe.db.whois.common.ip.Ipv4Resource;
+import net.ripe.db.whois.common.ip.Ipv6Resource;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TopLevelFilter<T extends Interval<T>> {
```

Comment:
We have this same class in whois-internal, should we move this class to whois-common and change the reference?

Reply:
Good idea